src/app/compiler.js

```diff
@@ -1,18 +1,13 @@
 var webworkify = require('webworkify');
-var queryParams = require('./query-params');
 var utils = require('./utils');
-var Renderer = require('./renderer');
 
 var Base64 = require('js-base64').Base64;
 
-function Compiler (editor, handleGithubCall, outputField, hidingRHP, updateFiles) {
-  var renderer = new Renderer(editor, this, updateFiles);
-
+function Compiler (editor, renderer, queryParams, handleGithubCall, outputField, hidingRHP, updateFiles) {
```

Comment:
Uff, these dependencies need to be slimmed down at some point providing just a set of callbacks I would say.